common/achievement/template.ts

```diff
@@ -62,4 +63,11 @@ async function getTemplatesByMetrics(metricsForAction: Metric[]) {
     return templatesForAction;
 }
 
-export { getAchievementTemplates, getTemplatesByMetrics };
+async function getTemplatesWithCourseRelation(): Promise<Achievement_template[]> {
+    const templatesByCourseRelation = await getAchievementTemplates(TemplateSelectEnum.BY_COURSE_RELATION);
+    const courseTemplates: Achievement_template[] = templatesByCourseRelation ? Array.from(templatesByCourseRelation['Subcourse']) : [];
```

Comment:
I think we should use the prisma enum here.
```suggestion
    const courseTemplates: Achievement_template[] = templatesByCourseRelation ? Array.from(templatesByCourseRelation[achievement_template_for_enum.Course]) : [];
```